libcsl_godot/src/ledger/transaction.rs

```diff
@@ -190,12 +259,96 @@ impl Utxo {
             ),
         )
     }
+
+    pub fn add_to_inputs_builder(&self, inputs_builder: &mut TxInputsBuilder) {
+        inputs_builder.add_key_input(
+            &CSL::address::BaseAddress::from_address(
+                &CSL::address::Address::from_bech32(&self.address.to_string()).unwrap(),
+            )
+            .unwrap()
+            .stake_cred()
+            .to_keyhash()
+            .unwrap(),
+            &CSL::TransactionInput::new(
+                &CSL::crypto::TransactionHash::from_hex(&self.tx_hash.to_string())
+                    .expect("Could not decode transaction hash"),
+                self.output_index,
+            ),
+            &Value::new_with_assets(
+                &to_bignum(
+                    self.coin
+                        .bind()
+                        .b
+                        .as_u64()
+                        .expect("UTxO Lovelace exceeds maximum")
+                        .into(),
+                ),
+                &multiasset_from_dictionary(&self.assets),
+            ),
+        );
+    }
+}
+
+
+#[derive(GodotClass)]
+#[class(base=RefCounted, rename=_CostModels)]
+pub struct CostModels {
+    pub cost_models: CSL::plutus::Costmdls
+}
+
+#[godot_api]
+impl CostModels {
+    #[func]
+    pub fn create() -> Gd<CostModels> {
+        Gd::from_object(CostModels { cost_models: CSL::plutus::Costmdls::new() })
+    }
+
+    fn build_model(ops: Array<u64>) -> CSL::plutus::CostModel {
+        let mut model = CSL::plutus::CostModel::new();
+        let mut i = 0;
+        for op in ops.iter_shared() {
```

Comment:
Small nitpick: here it would be more idiomatic to use [enumerate](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.enumerate) on the iterator instead of manually increasing `i` counter.